scripts/zones/VeLugannon_Palace/mobs/Zipacna.lua

```diff
@@ -4,122 +4,523 @@
+    { x = 63, y = 0, z = 456 },
+    { x = 60, y = 0, z = 450 },
+    { x = 60, y = 0, z = 443 },
+    { x = 60, y = 0, z = 419 },
+    { x = 40, y = 0, z = 419 },
+    { x = 21, y = 0, z = 419 },
+    { x = -9, y = 0, z = 420 },
+    { x = -59, y = 0, z = 420 },
+    { x = -60, y = 0, z = 433 },
+    { x = -60, y = 0, z = 450 },
+    { x = -62, y = 0, z = 455 },
+    { x = -67, y = 0, z = 458 },
+    { x = -75, y = 0, z = 460 },
+    { x = -76, y = 0, z = 460 },
+}
+
+    -- Path from the middle blue door to the first blue door
+local middleBlueToFirstBlue =
+{
+    { x = -82, y = 0, z = 460 },
+    { x = -93, y = 0, z = 459 },
+    { x = -103, y = 0, z = 459 },
+    { x = -106, y = 0, z = 458 },
+    { x = -110, y = 0, z = 454 },
+    { x = -116, y = 0, z = 434 },
+    { x = -117, y = 0, z = 420 },
+    { x = -121, y = 0, z = 409 },
+    { x = -125, y = 0, z = 395 },
+    { x = -129, y = 0, z = 385 },
+    { x = -133, y = 0, z = 381 },
+    { x = -141, y = 0, z = 380 },
+    { x = -153, y = 0, z = 379 },
+    { x = -157, y = 0, z = 379 },-- Middle blue door
+
+}
 entity.onMobSpawn = function(mob)
+    local pos = mob:getPos()
     mob:setMod(xi.mod.REGAIN, 200)
-    mob:pathThrough(pathNodes, xi.path.flag.PATROL)
+
+    -- Zip popped on blue short side (!pos -196, 0, 396)
+    if pos.x <= -195 then
+        mob:pathThrough(blueShortSideSpawnPath, xi.path.flag.COORDS)
+
+    -- Zip popped at the blue basement spawn point (!pos -121, 16, 420)
+    elseif pos.x <= -120 then
+        mob:pathThrough(blueBasementSpawnPath, xi.path.flag.COORDS)
+
+    -- Zip popped at the Yellow Short Side Spawn Point - going to first yellow door (!pos 191, 0, 399)
+    elseif pos.x >= 190 then
+        mob:pathThrough(yellowShortSideSpawnPath, xi.path.flag.COORDS)
+
+    -- Zip popped in the yellow side basement - going to first yellow door
+    elseif pos.x >= 140 then
+        mob:pathThrough(yellowBasementSpawnPath, xi.path.flag.COORDS)
+    end
+end
+
+entity.onMobFight = function(mob, target)
 end
 
 entity.onMobDeath = function(mob, player, optParams)
 end
 
+entity.onMobRoam = function(mob)
+    local pos              = mob:getPos()
+    local firstBlueDoor    = GetNPCByID(ID.npc.B_DOOR_OFFSET + 1)
+    local middleBlueDoor   = GetNPCByID(ID.npc.B_DOOR_OFFSET)
+    local firstYellowDoor  = GetNPCByID(ID.npc.Y_DOOR_OFFSET + 1)
+    local middleYellowDoor = GetNPCByID(ID.npc.Y_DOOR_OFFSET)
+    local getLastDoor      = mob:getLocalVar('[Zipacna]lastDoor')
+
+    -- begin of door checks. System in place is to check distance betwen mob and the NPC door (mapped to IDs.lua in local firstBlueDoorCoord, etc)
+    if
+        mob:checkDistance(firstBlueDoorCoord.x, firstBlueDoorCoord.y, firstBlueDoorCoord.z) <= 5
+    then
+        if -- POS from path to first blue door if Zip is coming from the basement
+            pos.x <= -162.000 and
+            pos.y <= 1 and
+            pos.z >= 379 and
+            getLastDoor ~= 1
+        then
```

Comment:
I'm struggling to understand what you're checking here?

You check the distance to the door (you can be using `mob:checkDistance(firstBlueDoor) <= 5`), and then you're checking some other point? Can you extract that to something named? Is it a part of one of the paths?

Then you're choosing which path to do depending on whether or not the door is closed. Fine, but the "if path is closed" is one other other other paths in reverse, no? I asked you in ASB or HXI discord to look at the `xi.path.flag.REVERSE` flat that `Guivre` uses so you don't have to have so much path data.

Reply:
If I am understanding your question correctly, there needs to be a check on both sides of the door. The reverse paths can work for the path between middle doors to the first doors but will not for the basement routes. I did not include them as I thought it would be confusing. Regardless, I feel all the paths would still need to be there due to coming from one side or the other. 

Maybe my approach is not the best way for this - thinking on it, I might be able to first check the status of the door (open/closed) and then determine which side of the door Zipacna is on. And from there, path accordingly.

Reply:
Do you even need to know which side of the door you're on? You already know which path you're using so if you're within whatever range of the door and it's closed, you know to reverse the current path. The side of approach is irrelevant because you already know which path you're using. If you don't, you can put your paths in a table and keep track of the index as a local var

Reply:
Helping Hook out with this - So will comment on some of this on his behalf.
Completely changed the door logic and used a local var (scoped to the lua) to determine the direction Zip is going.
Also updated to use `checkDistance` calls and `xi.path.flag.REVERSE` and shortened his pathing node list.
The pathing node list has also been converted to a table.